src/Providers/OpenAI/src/OpenAiProvider.cs

```diff
@@ -19,16 +19,18 @@ public class OpenAiProvider : Provider
 
     #region Constructors
 
```

Comment:
```suggestion

    public OpenAiProvider(OpenAIClient openAiClient)
        : base(id: OpenAiConfiguration.SectionName)
    {
        Api = openAiClient ?? throw new ArgumentNullException(nameof(openAiClient));
    }

```

Reply:
I suggest adding another constructor, for those who need it and it does not use the configuration

Reply:
In this case, properties from the parent Provider class (ChatSettings, EmbeddingSettings, etc.) will not be initialized because they have an 'init' setter.

Reply:
Do you mean code like this? It should work with init
```cs
new OpenAiProvider(new OpenAIClient())
{
    ChatSettings = new OpenAiChatSettings()
}
```

Reply:
But maybe we need to remove init restrictions, it doesn't make much sense to prohibit changes to the provider

Reply:
> Do you mean code like this? It should work with init
> 
> ```cs
> new OpenAiProvider(new OpenAIClient())
> {
>     ChatSettings = new OpenAiChatSettings()
> }
> ```

Oh... I forgot about that option. I guess in that case, your suggestion will do the trick.